tools/fastq_filtration.py

```diff
@@ -0,0 +1,117 @@
+    '@SRX079804:1:SRR292678:1:1101:190845:190845': ('CCTCAGCGTGGATTGCCGCTCATGCAGGAGCAGATAATCCCTTCGCCATCCCATTAAGCGCCGTTGTCGGTATTCC', 'FF@FFCFEECEBEC@@BBBBDFBBFFDFFEFFEB8FFFFFFFFEFCEB/>BBA@AFFFEEEEECE;ACD@DBBEEE'),
+    '@SRX079804:1:SRR292678:1:1101:198993:198993': ('AGTTATTTATGCATCATTCTCATGTATGAGCCAACAAGATAGTACAAGTTTTATTGCTATGAGTTCAGTACAACA', '<<<=;@B??@<>@><48876EADEG6B<A@*;398@.=BB<7:>.BB@.?+98204<:<>@?A=@EFEFFFEEFB'),
+    '@SRX079804:1:SRR292678:1:1101:204480:204480': ('AGTGAGACACCCCTGAACATTCCTAGTAAGACATCTTTGAATATTACTAGTTAGCCACACTTTAAAATGACCCG', '<98;<@@@:@CD@BCCDD=DBBCEBBAAA@9???@BCDBCGF=GEGDFGDBEEEEEFFFF=EDEE=DCD@@BBC')
+    } 
+
+def filter_by_quality(seq_and_quality:tuple, quality_threshold:int) -> str:
+    """
+    Filter sequence by quality. Decoding quality sequence in q-score list and calculate quality. 
+
+    Arguments:
+    - seq_and_quality (tuple): pair of fastq-seq and quality sequence
+    - quality_threshold (int): allow filter sequence and it is the bottom bound. Has got default value: 0. 
+
+    Return:
+    - str, if fastq-seq corresponds filtration condition.
+    - None in opposite situation.
+    """
+    seq = seq_and_quality[0]
+    quality_seq = seq_and_quality[1]
+
+    q_score = []
+    for symbol in quality_seq:
+        q_score.append(ord(symbol)-33)
+
+    quality = round(sum(q_score)/len(q_score), 4)
+    if quality >= quality_threshold:
+        return seq
+
+
+def filtre_by_length(seq:str, length_bounds:tuple) -> str:
+    """
+    Filter sequence by length.
+
+    Arguments:
+    - seq (str): fastq-seq
+    - length_bounds (tuple): bottom and top bound to filtration. Has got default value: (0, 2**32).
+
+    Return:
+    - str, if fastq-seq corresponds filtration condition.
+    - None in opposite situation.
+    """
+    length_seq = len(seq)
+    if length_seq >= length_bounds[0] and length_seq <= length_bounds[1]:
+        return seq
+
+
+
+def filter_by_gc_level(seq:str, gc_bounds:tuple) -> str:
+    """
+    Filter sequence by gc-level.
+
+    Arguments:
+    - seq (str): fastq-seq
+    - gc_bounds (tuple): bottom and top bound to filtration. Use only top bound if gc_bound is int.
+
+    Return:
+    - str, if fastq-seq corresponds filtration condition.
+    - None in opposite situation. 
+    """
+    if type(gc_bounds) == int:
+        bottom_bound = 0
+        top_bound = gc_bounds
+    elif type(gc_bounds) == tuple:
+        bottom_bound = gc_bounds[0]
+        top_bound = gc_bounds[1]
+
+    gc_amount = 0
+    for nucl in seq:
+        if nucl == "G" or nucl == "C":
+            gc_amount += 1
+    
+    cur_gc_level = round(gc_amount / len(seq) * 100, 4)
+    if cur_gc_level >= bottom_bound and cur_gc_level <= top_bound:
+        return seq
+
+
+    
+
+def filter_fastq(dict_of_fastq:dict, gc_bounds:tuple, length_bounds=(0, 2**32), quality_threshold=0) -> list:
```

Comment:
```suggestion
def filter_fastq(fastqs:dict, gc_bounds:tuple, length_bounds=(0, 2**32), quality_threshold=0) -> list:
```

1) Не надо начинать название переменной с ее типа данных). Вообще для этого ест ьаннотации типов.
2) Что-то сломалось. у границ gc есть только аннотация, а у границ по длине и качеству только значение по умолчанию. Так то бы везде и то и то дать:)

3) А оно точно возвращает list?